airflow/models/dag.py

```diff
@@ -533,94 +537,54 @@ def next_dagrun_info(
             "automated" DagRuns for this dag (scheduled or backfill, but not
             manual)
         """
-        if (
-            self.schedule_interval == "@once" and date_last_automated_dagrun
-        ) or self.schedule_interval is None:
-            # Manual trigger, or already created the run for @once, can short circuit
+        # XXX: The timezone.coerce_datetime calls in this function should not
+        # be necessary since the function annotation suggests it only accepts
+        # pendulum.DateTime, and someone is passing datetime.datetime into this
+        # function. We should fix whatever is doing that.
+        if self.is_subdag:
             return (None, None)
-        next_execution_date = self.next_dagrun_after_date(date_last_automated_dagrun)
-
-        if next_execution_date is None:
+        time_table: TimeTable = self.time_table
```

Comment:
Is the type needed here? The property already specifies the type 🤔

Reply:
Mypy cannot detect the type for `cached_property` for some reason. Not sure if it's because we're not 3.8+ or some other reasons, but this becomes `Any` if I don't declare the type.

Reply:
@uranusjr you can probably annotate `cached_property` with a generic type variable to work around this.

Reply:
https://github.com/python/mypy/pull/10408 might help when it's released, or something along the lines of https://github.com/python/mypy/issues/5858

Reply:
I think python/mypy#10408 is the one. Let me try the generic annotation method after the `compat` package is in. I'm thinking something like

```python
if TYPE_CHECKING:
    T = TypeVar("T")

    def cached_property(f: Callable[..., T]) -> T:
        ...
```

Reply:
I put the above in a `.pyi` file instead. Seems to pleas Mypy enough (and avoids Pylint annoyingly complaining about type stubs not conforming to Python coding styles).